bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckWorkItem.java

```diff
@@ -382,6 +389,20 @@ public Collection<WorkItem> prRun(Path scratchPath) {
                 return List.of(new CheckWorkItem(bot, prId, errorHandler, prUpdatedAt));
             }
 
+            // Check Force push
```

Comment:
I'm not sure if we should put this check here in `CheckWorkItem`. This will cause it to be run quite often. Maybe it would be better to put it in `CheckRun` where it would be guarded by the check metadata condition, so it would only need to be run if something significant has changed since the last check.

Reply:
> I'm not sure if we should put this check here in `CheckWorkItem`. This will cause it to be run quite often. Maybe it would be better to put it in `CheckRun` where it would be guarded by the check metadata condition, so it would only need to be run if something significant has changed since the last check.

I think this is already guarded by the check metadata condition. It will only be executed when `currentCheckValid(census, comments, activeReviews, labels)` is false. The if clause is very long(line 273-425).

Reply:
You are right, by bad. I misremembered the location of currentCheckValid. This looks like the right place to do this then.